### DISCUSSION TITLE
SNES and Mega Drive Recieving Bad CRC's

### DISCUSSION THREAD

**Discussion by majorpbx** [sanni/cartreader#153] [Need help]
Hello, I've built a Cart Reader and I've found that SNES and Mega Drive cartridges dump with CRC errors. I'm running version 6.1 if that helps. I'm going to test NES and Sega Master System but I'm guessing they may not work as well. Any help would be greatly appreciated.

What's Working:
N64 Cartridges/Saves
Gameboy Cartridges/Saves
Gameboy Color Cartridges/Saves
Gameboy Advanced Cartridges/Saves

What Isn't Working:
SNES Cartridges/Saves
Mega Drive

What I've tried:
Cleaned cartridge contacts
Reflow all solder connections
double check no bridges exist on any neighboring pins
2 different Arduino Mega 2560's

**Comment by sanni** [comment]
Dump a SNES game, copy it to your PC, open it together with a known good dump of the same game and version in HxD Hex Editor, do a CTRL+K compare. See at which address the differences start. That should give you an indication what address pins is the issue.
From your tests so far I would suspect SNES address pins A16-A23. So in HxD the differences should start somewhere after offset 0xFFFF. Could be a pin shorted to the GND plane.

**Reply by majorpbx** [reply]
Here's the differences.

![2021-05-17_8-57-39](https://user-images.githubusercontent.com/18235905/118492551-1f633880-b6ee-11eb-8e12-5ce251b4367d.jpg)

**Comment by sanni** [comment]
So from 0x0 to 0x7FFF it's okay, then it goes bad. Meaning SNES maskrom A0-A14 are fine and A15 is the issue. I guess you dumped a lorom game? Which would translate to A16 on the cartridge slot because of the way lorom is mapped.
Do you have a hirom cartridge too? With hirom the matched part should end at count 65536 (base 10).

If yes then this trace should be the issue:

![7979797431479](https://user-images.githubusercontent.com/19169494/118495077-1e98c980-b723-11eb-990e-e891bc836547.jpg)

**Reply by majorpbx** [reply]
Well there was an issue with A17 pin on the MD connector that I fixed so to speak. I also inspected the entire trace and found no issues and got .4-.5Ω from the back of the Arduino connector (cart reader side) to the pin on the connector. So now the MD dumped successfully but SNES still has the same issue.

Also, I don't believe I have a Hi ROM cartridge.

**Reply by majorpbx** [reply]
Checked traces for resistance for pins A15-A23 back to the Arduino connection and got .4-.5Ω for all. Still same issue with SNES dumps.

**Comment by sanni** [comment]
You can check if any pins are connected to GND that should not be connected.

**Reply by majorpbx** [reply]
Okay, secured a Hi ROM and here are the results. I'll keep checking each pin on the SNES connector.

![2021-05-17_21-44-45](https://user-images.githubusercontent.com/18235905/118577930-4014a800-b759-11eb-9786-d5ee3661568f.jpg)

**Comment by sanni** [comment]
Still the same pin making issues, probably shorted to GND. Here is a picture where the GND plane is highlighted:

![499494646464111](https://user-images.githubusercontent.com/19169494/118635739-38461980-b7d4-11eb-86b7-ec86072a002a.jpg)

On the right side the GND plane is pretty close to the pin, might be that the solder mask was damaged and a solder bridge was formed between the GND plane and the pin. 
Inspect the pin using a magnifying glass and remove the solder with solder wick to see if there is a short there.

**Reply by majorpbx** [reply]
After extensive testing, turns out that the pins for the MD and SNES connector were not making a sufficient connection. Everything else was testing good and didn't even think there might be a loose contact.
